src/lib/util/bitstream.h

```diff
@@ -207,11 +231,19 @@ inline void bitstream_out::write(uint32_t newbits, int numbits)
 			m_bits -= 8;
 		}
 
-	// shift the bits to the top
-	if (numbits == 0)
-		newbits = 0;
-	else
-		newbits <<= 32 - numbits;
+		// offload more bits if it'll still overflow the buffer
+		if (m_bits + numbits >= 32)
```

Comment:
This is the exact case this is fixing. The first write of 0x0f successfully writes 0x0f, but the second write of 0x0f only writes 0x0e and drops the last bit because it ended up where m_bits = 7 and the number of bits we wanted to write was 26, which is 33 bits total so the last bit was silently dropped.

```
m_buffer[42047d00] m_bits[29] newbits[0000000f] numbits[26] requiredbits[55] (before offloading to m_write buffer)
m_buffer[00000000] m_bits[ 5] numbits[26] requiredbits[31] (after offloading to m_write buffer)
newbits[000003c0] (after shifting newbits by 32 - numbits)
m_buffer[0000001e] m_bits[31] val[0000000f] (state at end of write func)

m_buffer[0000001e] m_bits[31] newbits[0000000f] numbits[26] requiredbits[57] (before offloading to m_write buffer)
m_buffer[1e000000] m_bits[ 7] numbits[26] requiredbits[33] (after offloading to m_write buffer)
newbits[000003c0] (after shifting newbits by 32 - numbits)
m_buffer[1e000007] m_bits[33] val[0000000e] (state at end of write func)
```